src/python_be.cc

```diff
@@ -1467,9 +1483,9 @@ ModelState::LaunchAutoCompleteStubProcess()
     RETURN_IF_ERROR(Stub()->Launch());
   }
   catch (const BackendModelException& ex) {
-    Stub()->TerminateStub(nullptr, nullptr, nullptr);
-    // Need to make sure that the shared memory pool is destructed properly
-    Stub()->ShmPool().reset();
+    Stub()->UpdateHealth();
+    Stub()->TerminateStub();
```

Comment:
I was wondering wondering why do we need to update health here?

Reply:
Because we have this condition [here](https://github.com/triton-inference-server/python_backend/blob/krish-python/src/stub_launcher.cc#L482) inside `TerminateStub()` that requires the updated health variable.